### PR TITLE
Beadsort implementation in C++

### LINKED ISSUE #1002: Bead Sort in C++
Add your implementation of Bead Sort Algorithm in C++.

### PR DESCRIPTION
**Pull request by Arunkumar0258**
Fixes #1002 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added a C++ implementation for Bead sort with time complexity in range of O(1) and O(S), where S is sum of all elements in array.

#### Languages Used:
- C++

#### Files Added:
- Bead_Sort.cpp

### CODE REVIEW

**Comment by GOVINDDIXIT** [comment]
@Arunkumar0258 You can take help from various sites but it will be good for you if you write it on your own.

**Comment by Arunkumar0258** [comment]
@somya-kapoor @GOVINDDIXIT  Thanks for the review. I will try to optimize the code further and get back to you.

**Comment by Arunkumar0258** [comment]
@somya-kapoor  I have added vector implementation with some C++ optimizations I learnt recently. Please review.

File: Bead_Sort/Bead_Sort.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{
+    // initialize vector
+    vector<int> arr = {23, 2, 12, 54, 90, 102, 32};
+    int n = sizeof(arr) / sizeof(arr[0]);
+
+    // reserve static vector for optimal performance gain
+    arr.reserve(n);
+
+    // Print vector before sorting
+    cout<< "Before sorting" <<endl;
```
Comment (somya-kapoor) [inline Bead_Sort/Bead_Sort.cpp]: Please add space before and after <<.

File: Bead_Sort/Bead_Sort.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    for(auto const& i: arr)
+        cout << i <<" ";
+    cout <<endl;
```
Comment (somya-kapoor) [inline Bead_Sort/Bead_Sort.cpp]: Please add space.

File: Bead_Sort/Bead_Sort.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    beadSort(arr);
+
+    // Print vector after sorting
+    cout<< "After sorting" <<endl;
+    for(auto const& i: arr)
+        cout << i <<" ";
+    cout <<endl;
```
Comment (somya-kapoor) [inline Bead_Sort/Bead_Sort.cpp]: Please add space

File: Bead_Sort/Bead_Sort.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    // Print vector before sorting
+    cout<< "Before sorting" <<endl;
+    for(auto const& i: arr)
+        cout << i <<" ";
+    cout <<endl;
+
+    beadSort(arr);
+
+    // Print vector after sorting
+    cout<< "After sorting" <<endl;
```
Comment (somya-kapoor) [inline Bead_Sort/Bead_Sort.cpp]: Please add space

File: Bead_Sort/Bead_Sort.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    for(auto const& i: arr)
+        cout << i <<" ";
+    cout <<endl;
+
+    // Free memory space of vector
+    arr.shrink_to_fit();
+}
```
Comment (somya-kapoor) [inline Bead_Sort/Bead_Sort.cpp]: Please add output.

**Comment by Arunkumar0258** [comment]
@somya-kapoor I have incorporated the requested changes. Please review.

**Comment by somya-kapoor** [comment]
@Arunkumar0258 Good work. Last thing, Please squash all your commits.

**Comment by Arunkumar0258** [comment]
@somya-kapoor I think I have squashed all my commits. Please see.

**Comment by somya-kapoor** [comment]
@Arunkumar0258 No, your commits are not squashed. Have a look at https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git.
If you still face the issue, please connect on slack channel. Will be happy to help. :)

**Comment by Arunkumar0258** [comment]
@somya-kapoor I have squashed the commits. Please review.

**Comment by somya-kapoor** [comment]
Good work @Arunkumar0258 .
@jainaman224 It is approved. You can have a final look.

**Pull request merged**